### PR TITLE
blitz: fix for fujitsu compiler

### PR DESCRIPTION
**Pull request by kinagaki-fj**
Added a patch and autoreconf function for fujitsu compiler

### CODE REVIEW

File: var/spack/repos/builtin/packages/blitz/package.py [hunk 1]
```diff
@@ -17,6 +17,15 @@ class Blitz(AutotoolsPackage):
     version("1.0.2", sha256="500db9c3b2617e1f03d0e548977aec10d36811ba1c43bb5ef250c0e3853ae1c2")
 
     depends_on("python@3:", type="build")
+    depends_on("m4", type="build")
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+
+    patch("fujitsu_compiler_specfic_header.patch", when="%fj")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/blitz/package.py:26]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

File: var/spack/repos/builtin/packages/blitz/package.py [hunk 2]
```diff
@@ -17,6 +17,15 @@ class Blitz(AutotoolsPackage):
...
+
+    def autoreconf(self, spec, prefix):
+        autoreconf("--install", "--verbose", "--force")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/blitz/package.py]: This should be the default. Why was it necessary to add this function?

**Review by alalazo — CHANGES_REQUESTED** [review]
Mostly LGTM. Only two minor comments.

File: var/spack/repos/builtin/packages/blitz/package.py (hunk 1 above)
Comment (kinagaki-fj) [reply var/spack/repos/builtin/packages/blitz/package.py:26]: @alalazo 
I have added a comennt. This patch fixes makefile and include header file to build with Fujitsu compiler.

File: var/spack/repos/builtin/packages/blitz/package.py (hunk 2 above)
Comment (kinagaki-fj) [reply var/spack/repos/builtin/packages/blitz/package.py]: @alalazo 
Fixed. This function currently is removed now.

**Review by alalazo — APPROVED** [review]

**Pull request merged**